### PR TITLE
[docs-only] Update the docs helper readme and other related files

### PR DESCRIPTION
**Pull request by mmattel**
Referencing: #8454 (Envar delta reflections)

As discussed with @dragonchaser 

Post merging the envvar delta reflections, we need to update the readme to document how and when to use the reflection go code. Additionally, updated and fixed envvar related yaml files have been added as they got updated by the process too.

As next step, I will provide the required adoc template files for added/removed and deprecated envvars to finalize the automation process.

@phil-davis mind to do a language review?

### CODE REVIEW

File: docs/helpers/README.md [hunk 1]
```diff
@@ -12,16 +12,17 @@
       * [General Extended Envvars Info](#general-extended-envvars-info)
       * [Extract Extended Envvars](#extract-extended-envvars)
       * [Generate Extended Envvar Docs](#generate-extended-envvar-docs)
+   * [Tasks for New Releases](#tasks-for-new-releases)
    * [Backporting](#backporting)
 
 ## Introduction
 
-`docs/helpers` contains small go programs creating docs by extracting information from the code. The `main.go` program is manually started with `make docs-generate` or via the CI. It calls the other required programs and has these main responsibilities:
+`docs/helpers` contains go programs creating docs by extracting information from the code. The `main.go` program is manually started with `make docs-generate` or via the CI. It calls the other required programs and has these main responsibilities:
```
Comment (dragonchaser) [inline docs/helpers/README.md:20]: it is just one program actually

**Review by dragonchaser — APPROVED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
-#  documentation
+#  Documentation
 
 To contribute to documentation please see also: [Documentation](http://owncloud.dev/ocis/development/build-docs/)
+
+Note that when running a helper like `make -C docs docs-generate` locally, you will see the generated output in e.g. docs/services/<service-name>. When merging a PR, the pipeline will generate the output in the master branch but finally moves it into the `docs` branch. You will therefor _not_ see changes in the docs folder of the master branch!
```
Comment (phil-davis) [inline docs/README.md:5]:
```suggestion
Note that when running a helper like `make -C docs docs-generate` locally, you will see the generated output in e.g. docs/services/<service-name>. When merging a PR, the pipeline will generate the output in the master branch but finally moves it into the `docs` branch. You will therefore _not_ see changes in the docs folder of the master branch!
```

File: docs/helpers/README.md [hunk 3]
```diff
@@ -12,16 +12,17 @@
       * [General Extended Envvars Info](#general-extended-envvars-info)
       * [Extract Extended Envvars](#extract-extended-envvars)
       * [Generate Extended Envvar Docs](#generate-extended-envvar-docs)
+   * [Tasks for New Releases](#tasks-for-new-releases)
    * [Backporting](#backporting)
 
 ## Introduction
 
-`docs/helpers` contains small go programs creating docs by extracting information from the code. The `main.go` program is manually started with `make docs-generate` or via the CI. It calls the other required programs and has these main responsibilities:
+`docs/helpers` contains a go program named `main.go` which creates docs by extracting information from the code using additional go programs. Individual steps (programs) can be called manually if needed. Note that not all programs are called automatically by purpose, see the [Tasks for New Releases](#tasks-for-new-releases) below. `main.go` is used by `make docs-generate` which can be called manually or by the CI. It calls the other required programs and has these main responsibilities for automatic runs:
```
Comment (phil-davis) [inline docs/helpers/README.md:20]:
```suggestion
`docs/helpers` contains a go program named `main.go` which creates docs by extracting information from the code using additional go programs. Individual steps (programs) can be called manually if needed. Note that not all programs are called automatically on purpose, see the [Tasks for New Releases](#tasks-for-new-releases) below. `main.go` is used by `make docs-generate` which can be called manually or by the CI. It calls the other required programs and has these main responsibilities for automatic runs:
```

File: docs/helpers/README.md [hunk 4]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
-IMPORTANT: `RawName`, `Path` and `FoundInCode` are automatically filled by the program. DO NOT EDIT THESE VALUES MANUALLY.
+**IMPORTANT**: `RawName`, `Path` and `FoundInCode` are automatically filled by the program. DO NOT EDIT THESE VALUES MANUALLY.
 
 ### Generate Extended Envvar Docs
 
 The process further picks up the `yaml` file generated in the `Extract Rogue Envvars` step and renders it to an adoc file (a table is created) using a go template. The template file for this step can be found at `docs/templates/ADOC_extended.tmpl`.
 
+## Tasks for New Releases
+
+Close before a new release gets published, but **before** a new ocis admin docs branch is created, some tasks need to be done manually. These tasks cant be done automatically! By processing these manual tasks, doc related and important files get updated respectively created, ready to be consumed by the admin docs. The following situations can occur, see the solution provided:
```
Comment (phil-davis) [inline docs/helpers/README.md:155]:
```suggestion
Close before a new release gets published, but **before** a new ocis admin docs branch is created, some tasks need to be done manually. These tasks cant be done automatically! By processing these manual tasks, doc related and important files get updated or created, ready to be consumed by the admin docs. The following situations can occur, see the solution provided:
```

File: docs/helpers/README.md [hunk 5]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+
+1. Admin docs prepares in master for envvar delta changes. Building the admin docs report inexisting referenced delta files.\
```
Comment (phil-davis) [inline docs/helpers/README.md:157]:
```suggestion
1. Admin docs prepares in master for envvar delta changes. Building the admin docs report non-existent referenced delta files.\
```

File: docs/helpers/README.md [hunk 6]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+&#8594; Run the manual tasks described below and restart the branched admin docs.
+
+2. A new admin docs branch gets created but no `docs-stable-x.y` branch has been created in the ocis repo so far. Building the branched admin docs report inexisting referenced files.\
```
Comment (phil-davis) [inline docs/helpers/README.md:160]:
```suggestion
2. A new admin docs branch gets created but no `docs-stable-x.y` branch has been created in the ocis repo so far. Building the branched admin docs reports non-existent referenced files.\
```

File: docs/helpers/README.md [hunk 7]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+&#8594; Run the manual tasks described below.\
+&#8594; Take care that the `docs-stable-x.y` branch gets created in the ocis repo.\
+&#8594; Alternatively you can _temporarily_ point in the branched [docs-ocis/antora.yml](https://github.com/owncloud/docs-ocis/blob/master/antora.yml) `compose_tab_1` to `master`.\
+&#8594; Restart building the branched admin docs.
+
+3. A new ocis release is close to be released, the manual tasks **have not been** processed so far.\
```
Comment (phil-davis) [inline docs/helpers/README.md:166]:
```suggestion
3. A new ocis release is close to being released, the manual tasks **have not been** processed so far.\
```

File: docs/helpers/README.md [hunk 8]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+&#8594; Run the manual tasks AND backport the results into the `docs-stable-x.y` branch.
+
+4. A new ocis release is close to be released, the manual tasks **have been** processed some time ago.\
```
Comment (phil-davis) [inline docs/helpers/README.md:169]:
```suggestion
4. A new ocis release is close to being released, the manual tasks **have been** processed some time ago.\
```

File: docs/helpers/README.md [hunk 9]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+&#8594; Re-run the manual tasks AND backport the results into the `docs-stable-x.y` branch.
+
+5. A new ocis release **has been released**, the manual tasks **have not been** processed so far.\
+&#8594; Re-run the manual tasks AND backport the results into the `docs-stable-x.y` branch.
+  
+### Task List
+
+1. From the ocis root, run `make -C docs docs-generate` and check if there is a change in the `extended-envars.yaml` output.\
+&#8594; In case, process [Extended Envvars](#extended-envvars). When done, re-run the make command and check if the output of `./docs/services/_includes/adoc/extended_configvars.adoc` matches the expectations.
```
Comment (phil-davis) [inline docs/helpers/README.md:178]:
```suggestion
&#8594; In this case, process [Extended Envvars](#extended-envvars). When done, re-run the make command and check if the output of `./docs/services/_includes/adoc/extended_configvars.adoc` matches the expectations.
```

File: docs/helpers/README.md [hunk 10]
```diff
@@ -140,20 +141,65 @@ type Variable struct {
...
+
+2. In `./docs/helpers` run: `go run . --help` This will give you an overview of available commands. 
+    1. Run `go run . all` to generate respectively update required base files.\
```
Comment (phil-davis) [inline docs/helpers/README.md:181]:
```suggestion
    1. Run `go run . all` to generate and/or update required base files.\
```

**Review by phil-davis — APPROVED** [review]

**Pull request merged**